src/main/java/gift/service/WishService.java

```diff
@@ -34,35 +36,52 @@ public WishResponseDTO addWish(WishRequestDTO wishRequestDTO, Member member) {
         return wishRepository.findByMemberIdAndProductId(member.getId(), productId)
             .map(existingWish -> {
                 int newQuantity = existingWish.getQuantity() + wishRequestDTO.quantity();
-                wishRepository.updateQuantity(member.getId(), productId, newQuantity);
+                existingWish.setQuantity(newQuantity);  // JPA 엔티티 기반 업데이트
+                wishRepository.save(existingWish);
                 return new WishResponseDTO(member.getId(), new ProductResponseDTO(product), newQuantity);
             })
             .orElseGet(() -> {
-                wishRepository.save(new Wish(member.getId(), productId, wishRequestDTO.quantity()));
+                wishRepository.save(new Wish(member, product, wishRequestDTO.quantity()));
                 return new WishResponseDTO(member.getId(), new ProductResponseDTO(product), wishRequestDTO.quantity());
             });
     }
 
     @Transactional(readOnly = true)
     public List<WishResponseDTO> getWishes(Member member, int page, int size, String sort) {
-        long offset = (long) page * size;
-        List<WishWithProduct> wishWithProducts = wishRepository.findByMemberIdWithPagination(member.getId(), size, offset, sort);
+        Sort sortBy;
+        switch (sort) {
+            case "name":
+                sortBy = Sort.by("product.name");
+                break;
+            case "price":
+                sortBy = Sort.by("product.price");
+                break;
+            default:
+                sortBy = Sort.by("id");
```

Comment:
[java의 switch expression](https://velog.io/@nunddu/Java-Switch-Expression-in-Java-14)을 활용해보면 조금 더 깔끔하게 변경해볼 수 있을 것 같아요.  😃 

```suggestion
        Sort sortBy = switch (sort) {
            case "name" -> Sort.by("product.name");
            case "price" -> Sort.by("product.price");
            default -> Sort.by("id");
        };
```

Reply:
Pageable은 사실상 2단계 미션인데, 이미 구현을 하신듯하군요..? 😅

Reply:
API 명세에 queryString으로 page=0&size=10&sort=createdDate,desc 같은 값이 들어가 있어서 구현해야 하는줄 알았네요..